spec/javascripts/helpers/fetch-mock.js

```diff
@@ -1,5 +1,8468 @@
 /* global fetchMock */
 
-//= require fetch-mock/es5/client-bundle.js
```

Comment:
This file seems to contain both the vendored code and a customisation (`beforeEach`) which seems a bit dangerous.

I'd suggest that we have separate files. I think it'd be a good idea to make use of a  `spec/javascripts/vendors` for vendored dependencies. I think that we could also move jasmine-jquery there.

Other thought is that it'd be good to record what version of the dependency we have (filename might be best way to express that - i.e. fetch-mock-9.11.0.js. Ideal would be a weblink we can download from that we can reference in a comment that people can use for future updates.

As an aside a whacky counter idea could be to pull the dependency in from a JS module e..g `import fetchMock from "view-source:https://ga.jspm.io/npm:fetch-mock@9.11.0/es5/client-bundle.js"` but rabbit hole / can of worms / dodgy assumptions

Reply:
I did consider the `import` approach – as I know this is something that can be done with Skypack: https://www.skypack.dev/view/fetch-mock

But if I understand it correctly, we'd need the script to be loaded as `<script type="module">`, which I don't think is the case. Very happy to be proven wrong on that, though.

Reply:
I understand it does do something different if you have an .mjs file - presumably turns into a module import.

I’d suggest having a quick test - but I’m expecting we’d find some sort of reason for a dead end. (I’m expecting if things did work we might still hit a scoping issue)

Reply:
@kevindew what do you think about this approach instead? 51d638b5d6ed52a1359bad7c57cfbf8b500b4ac2